numpy/ma/core.py

```diff
@@ -285,8 +285,8 @@ def _extremum_fill_value(obj, extremum, extremum_name):
     def _scalar_fill_value(dtype):
         try:
             return extremum[dtype]
-        except KeyError:
-            raise TypeError(f"Unsuitable type {dtype} for calculating {extremum_name}.")
+        except KeyError as e:
+            raise TypeError(f"Unsuitable type {dtype} for calculating {extremum_name}.") from e
```

Comment:
This one should also be `from None`. In general, things which catch only `KeyError` should just include the offending key in their message, rather than chaining tracebacks.